compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2810,16 +2811,21 @@ public void visit(BLangSimpleVarRef varRefExpr, AnalyzerData data) {
         }
 
         Name compUnitName = getCurrentCompUnit(varRefExpr);
-        varRefExpr.pkgSymbol =
-                symResolver.resolvePrefixSymbol(data.env, names.fromIdNode(varRefExpr.pkgAlias), compUnitName);
-        if (varRefExpr.pkgSymbol == symTable.notFoundSymbol) {
+        BSymbol pkgSymbol = symResolver.resolvePrefixSymbol(data.env, names.fromIdNode(varRefExpr.pkgAlias),
+                                                            compUnitName);
+        varRefExpr.pkgSymbol = pkgSymbol;
+        if (pkgSymbol == symTable.notFoundSymbol) {
             varRefExpr.symbol = symTable.notFoundSymbol;
             dlog.error(varRefExpr.pos, DiagnosticErrorCode.UNDEFINED_MODULE, varRefExpr.pkgAlias);
+        } else if (Names.CLIENT.equals(varName) &&
+                !identifier.isLiteral &&
+                !symTable.clientDeclarations.containsValue(pkgSymbol.pkgID)) {
+            dlog.error(identifier.pos, DiagnosticErrorCode.INVALID_USAGE_OF_THE_CLIENT_KEYWORD_AS_UNQUOTED_IDENTIFIER);
```

Comment:
Why do we need to disallow this? In `BLangIdentifier` we preserve `originalValue` which can be used to distinguish `module:'client` and `module:client`

Reply:
> Why do we need to disallow this?

`foo:client` has special semantics: it refers to a client object in the module generated for the client declaration, it doesn't necessarily mean it is referring to a client object named as `client`.

> In `BLangIdentifier` we preserve `originalValue` which can be used to distinguish `module:'client` and `module:client`

Did you mean as an alternative for `Names.CLIENT.equals(varName) && !identifier.isLiteral`? Original value seems to be skipped with qualified identifiers in https://github.com/ballerina-platform/ballerina-lang/pull/37471/files/1505e843c79d64fad5d8ef5f2a73149bce3a40b7#diff-e960b3c3a4848ff505626da90f021ee41db746cf49dcf6e6d69e56f84b23ad36R5336 which is why marking `isLiteral` there was required.

Reply:
if the user writes `foo:client` it refers to client object from client declaration if user writes `foo:'client` then it's the usual behavior. isn't it?

Reply:
This is okay. I have misunderstood previously